### PR TITLE
Deprecate Entrust Certificates

### PR DESCRIPTION
**Pull request by SidneyAllen**
In 2017, Xero will begin deprecation of entrust certificates for Partner Apps.  I've updates the base URL and removed the client_cert requirement and tested the example partner app.

You can read about this change in our Dev Center.
http://developer.xero.com/documentation/auth-and-limits/entrust-certificate-deprecation/

Thanks

### CODE REVIEW

**Comment by romgar** [comment]
I tested with these changes, works like a charm. Can be merged!

File: xero/auth.py [hunk 1]
```diff
@@ -336,7 +336,7 @@ class PartnerCredentials(PublicCredentials):
         oauth_authorization_expires_at tells when the overall access
         permissions expire (~10 year window)
     """
-    def __init__(self, consumer_key, consumer_secret, rsa_key, client_cert,
+    def __init__(self, consumer_key, consumer_secret, rsa_key, client_cert=None,
```
Comment (romgar) [inline xero/auth.py]: The docstring of this class should be updated regarding `client_cert`. It should say that this argument is no more needed regarding new Xero security policy, but stays for now to be backward-compatible.

**Comment by romgar** [comment]
@SidneyAllen getting a `The page you are attempting to access requires your browser to have a Secure Sockets Layer (SSL) client certificate that the Web server recognizes.` when trying to renew my access token. Any idea? 

As a side note, querying directly an endpoint (Organisations, Transactions, ...) works without SSL client certificate.

**Comment by avandehulsbeek** [comment]
@romgar FWIW, I haven't had any issues with this when renewing access tokens using this (running for a couple of days on an appengine instance).

**Comment by romgar** [comment]
@avandehulsbeek my bad, I missed some changes. Works like a charm now.

:+1: for this change! Any feedback @aidanlister @freakboy3742 ?

File: xero/auth.py (hunk 1 above)
Comment (SidneyAllen) [inline xero/auth.py]: added note about client_cert https://github.com/freakboy3742/pyxero/pull/168/commits/0b7a95e3ca5437a8bb991aef5c9af2a284021e22

**Comment by SidneyAllen** [comment]
Thanks @avandehulsbeek and @romgar 

Appreciate you testing this and looking forward to seeing this merged.

File: xero/auth.py [hunk 2]
```diff
@@ -318,8 +318,11 @@ class PartnerCredentials(PublicCredentials):
 
         >>> rsa_key = "-----BEGIN RSA PRIVATE KEY----- ..."
 
-     2) You'll need to pass a tuple to the Entrust certificate pair.
-
+     2) client_cert is no longer needed regarding new Xero security policy, 
+        but stays for now to be backward-compatible.
+        If you are still implementing Entrust Certificates you'll need
```
Comment (romgar) [inline xero/auth.py]: I don't think this comment is completely right. If people are using `pyxero` after this merge, they can't use anymore Entrust certificates pair. I think the first sentence should be something more like: `client_cert is no longer used regarding new Xero security policy, but stays for now for backward-compatibility.` and remove `If you are still [...] pair.`

Comment (SidneyAllen) [inline xero/auth.py]: done

**Comment by romgar** [comment]
Ready to be merged @aidanlister :+1:

**Pull request merged**